presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -71,7 +77,22 @@ public Analyzer(Session session,
 
     public Analysis analyze(Statement statement)
     {
-        return analyze(statement, false);
+        Analysis analysis = analyze(statement, false);
+        boolean snapshotTimestampUsed = analysis.getTables().stream().anyMatch(tableHandle -> tableHandle.getConnectorHandle().getSnapshotTimestampMS().isPresent());
```

Comment:
Analyzer should not check for connector specific details. If we want to check for snapshots here, we should either add it to `TableHandle` (support snapshot in presto core engine generically), or do this check somewhere else.